### PR TITLE
[CPU] Add SetBlob/GetBlob tests for dynamic shapes cases

### PR DESCRIPTION
**Pull request by egorshul**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *70934*

### CODE REVIEW

File: inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp [hunk 1]
```diff
@@ -176,6 +176,116 @@ TEST_P(OVInferRequestIOTensorTest, canInferWithGetOut) {
...
+    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+    OV_ASSERT_NO_THROW(req.set_tensor("input_tensor", tensor));
+    OV_ASSERT_NO_THROW(req.infer());
+    OV_ASSERT_NO_THROW(req.start_async());
+    OV_ASSERT_NO_THROW(req.wait());
+    ov::runtime::Tensor actual_tensor = req.get_tensor("input_tensor");
+    ASSERT_EQ(tensor.data(), actual_tensor.data());
+}
+
+TEST_P(OVInferRequestIOTensorSetTensorGetTensorDynamicShapesTest, setTensorDifferentLayoutsDynamicShapes) {
```
Comment (mandrono) [inline inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp:257]: Could you also test next scenario (or check that this tests already exist):
1. `get_tensor -> set_shape -> infer -> check results` for input/output
2. tensor with rank > 5 (blocked layout) and different stride: 1) planar - abcdef (should pass) 2) i.e. acdefb (should failed)
3. also we need to cover cases for output blob. 1) set output blob with expected shape after inference -> infer -> and check that output blob contain correct data 2) add the same as `setTensorDifferentLayoutsDynamicShapes` and point 2 for output blob

File: inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp [hunk 2]
```diff
@@ -176,6 +176,116 @@ TEST_P(OVInferRequestIOTensorTest, canInferWithGetOut) {
...
+    auto mult = ngraph::builder::makeEltwise(split->output(1), in2mult, ngraph::helpers::EltwiseTypes::MULTIPLY);
+    auto relu2 = std::make_shared<ngraph::opset1::Relu>(mult);
+
+    auto concat = std::make_shared<ngraph::opset1::Concat>(ngraph::OutputVector{relu1->output(0), relu2->output(0)}, 3);
+    concat->get_output_tensor(0).set_names({"concat"});
+
+    function = std::make_shared<ngraph::Function>(concat, params, "SplitAddConcat");
+}
+
+TEST_P(OVInferRequestIOTensorSetTensorGetTensorDynamicShapesTest, setTensorDynamicShapes) {
```
[the remark this answers is not in the document]
Comment (egorshul) [reply inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp:219]: Done

File: inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp (hunk 1 above)
Comment (egorshul) [reply inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp:348]: Added tests. Thank you.

File: inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp [hunk 3]
```diff
@@ -176,6 +176,116 @@ TEST_P(OVInferRequestIOTensorTest, canInferWithGetOut) {
...
+    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+
+    ov::runtime::Tensor tensor(ngraph::element::Type_t::f32, {2, 2, 15, 15});
+    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+    OV_ASSERT_NO_THROW(req.set_tensor("input_tensor", tensor));
+    OV_ASSERT_NO_THROW(req.infer());
+    OV_ASSERT_NO_THROW(req.start_async());
+    OV_ASSERT_NO_THROW(req.wait());
+    ov::runtime::Tensor actual_tensor = req.get_tensor("input_tensor");
+    ASSERT_EQ(tensor.data(), actual_tensor.data());
```
[the remark this answers is not in the document]
Comment (egorshul) [reply inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp]: Removed the test. The scenario described here is already being tested.

File: inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp [hunk 4]
```diff
@@ -176,6 +176,217 @@ TEST_P(OVInferRequestIOTensorTest, canInferWithGetOut) {
...
+    auto el = ngraph::element::Type_t::f32;
+
+    auto byteStrides = [](const ov::Strides& strides, const ov::element::Type& type) -> ov::Strides {
+        ov::Strides byte_strides(strides.size());
+        for (size_t i = 0; i < strides.size(); ++i)
+            byte_strides[i] = strides[i] * type.size();
+        return byte_strides;
+    };
+
+    ov::runtime::Tensor tensor1(el, {1, 4, 10, 10, 8, 16}, data, byteStrides({51200, 12800, 1280, 128, 16, 1}, el));
```
Comment (mandrono) [inline inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp:302]: is it possible to calculate strides instead "hardcode" values?

File: inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp [hunk 5]
```diff
@@ -176,6 +176,217 @@ TEST_P(OVInferRequestIOTensorTest, canInferWithGetOut) {
...
+    std::map<std::string, ov::PartialShape> shapes;
+    shapes[tensor_name] = {{1, 5}, {1, 5}, {10, 20}, {10, 20}};
+    OV_ASSERT_NO_THROW(function->reshape(shapes));
+    OV_ASSERT_NO_THROW(execNet = core->compile_model(function, targetDevice, configuration));
+    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+
+    ov::runtime::Tensor actualTensor = ov::test::utils::create_and_fill_tensor(element::f32, {1, 4, 20, 20}, 10, 1);
+    OV_ASSERT_NO_THROW(req.set_tensor("input_tensor", actualTensor));
+    ov::runtime::Tensor outTensor = req.get_tensor("ReLU");
+    OV_ASSERT_NO_THROW(outTensor.set_shape({1, 4, 20, 20}));
```
Comment (mandrono) [inline inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp:340]: I just understood that for output tensor we need test some different scenario
1. We need to set tensor with preallocated memory and check that after inference we store correct values in buffer
2. Call `set_tensor` for output

File: inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp [hunk 6]
```diff
@@ -176,6 +176,217 @@ TEST_P(OVInferRequestIOTensorTest, canInferWithGetOut) {
...
+    ov::runtime::Tensor outTensor = req.get_tensor("ReLU");
+    ASSERT_EQ(actualTensor.get_shape(), outTensor.get_shape());
+    for (int i = 0; i < actualTensor.get_size(); i++) {
+        if (actualTensor.data<float>()[i] >= 0)
+            ASSERT_EQ(actualTensor.data<float>()[i], outTensor.data<float>()[i]);
+        else
+            ASSERT_EQ(0, outTensor.data<float>()[i]);
+    }
+
+    ie::BlockingDesc blockingDesc2{{1, 4, 20, 20}, InferenceEngine::NHWC};
```
Comment (mandrono) [inline inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp:384]: just comments
here `BlockingDesc` also identical I guess

File: inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp [hunk 7]
```diff
@@ -176,6 +176,217 @@ TEST_P(OVInferRequestIOTensorTest, canInferWithGetOut) {
...
+
+    OV_ASSERT_NO_THROW(req.infer());
+
+    ASSERT_EQ(actualTensor.get_shape(), outTensor.get_shape());
+    ov::test::utils::compare(actualTensor, outTensor, std::numeric_limits<float>::max(), std::numeric_limits<float>::max());
+}
+
+TEST_P(OVInferRequestIOTensorSetTensorGetTensorDynamicShapesTest, setTensorDifferentLayoutsDynamicShapes) {
+    const std::string tensor_name = "input_tensor";
+    std::map<std::string, ov::PartialShape> shapes;
```
Comment (mandrono) [inline inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp:350]: what about coverage output tensors?

File: inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp [hunk 8]
```diff
@@ -176,6 +176,170 @@ TEST_P(OVInferRequestIOTensorTest, canInferWithGetOut) {
...
+        for (int i = 0; i < tensor.get_size(); i++) {
+            if (tensor.data<float>()[i] >= 0)
+                ASSERT_EQ(tensor.data<float>()[i], outTensor.data<float>()[i]);
+            else
+                ASSERT_EQ(0, outTensor.data<float>()[i]);
+        }
+    }
+}
+
+TEST_P(OVInferRequestIODynamicTensorTest, specialCase1) {
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp:238]: specialCase1 => to something meaningful

File: inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp [hunk 9]
```diff
@@ -176,6 +176,170 @@ TEST_P(OVInferRequestIOTensorTest, canInferWithGetOut) {
...
+
+        OV_ASSERT_NO_THROW(req.infer());
+
+        ov::runtime::Tensor outTensor = req.get_tensor("ReLU");
+        ASSERT_EQ(actualTensor.get_shape(), outTensor.get_shape());
+        for (int i = 0; i < actualTensor.get_size(); i++) {
+            if (actualTensor.data<float>()[i] >= 0)
+                ASSERT_EQ(actualTensor.data<float>()[i], outTensor.data<float>()[i]);
+            else
+                ASSERT_EQ(0, outTensor.data<float>()[i]);
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp:251]: why do we need this if >= 0 ?

File: inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp [hunk 10]
```diff
@@ -176,6 +176,170 @@ TEST_P(OVInferRequestIOTensorTest, canInferWithGetOut) {
...
+    auto el = ngraph::element::Type_t::f32;
+
+    auto byteStrides = [](const ov::Strides& strides, const ov::element::Type& type) -> ov::Strides {
+        ov::Strides byte_strides(strides.size());
+        for (size_t i = 0; i < strides.size(); ++i)
+            byte_strides[i] = strides[i] * type.size();
+        return byte_strides;
+    };
+
+    ie::BlockingDesc blockingDesc1{{1, 4, 10, 10, 8, 16}, {0, 1, 2, 3, 4, 5}};
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp:270]: why old API is used?

File: inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp [hunk 11]
```diff
@@ -176,6 +176,170 @@ TEST_P(OVInferRequestIOTensorTest, canInferWithGetOut) {
...
+        for (int i = 0; i < actualTensor.get_size(); i++) {
+            if (actualTensor.data<float>()[i] >= 0)
+                ASSERT_EQ(actualTensor.data<float>()[i], outTensor.data<float>()[i]);
+            else
+                ASSERT_EQ(0, outTensor.data<float>()[i]);
+        }
+    }
+}
+
+TEST_P(OVInferRequestIODynamicTensorTest, specialCase2) {
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp:256]: specialCase2 => proper name

File: inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp [hunk 12]
```diff
@@ -176,6 +176,170 @@ TEST_P(OVInferRequestIOTensorTest, canInferWithGetOut) {
...
+
+    auto byteStrides = [](const ov::Strides& strides, const ov::element::Type& type) -> ov::Strides {
+        ov::Strides byte_strides(strides.size());
+        for (size_t i = 0; i < strides.size(); ++i)
+            byte_strides[i] = strides[i] * type.size();
+        return byte_strides;
+    };
+
+    ie::BlockingDesc blockingDesc1{{1, 4, 10, 10, 8, 16}, {0, 1, 2, 3, 4, 5}};
+    ov::runtime::Tensor tensor1(el, blockingDesc1.getBlockDims(), data, byteStrides(blockingDesc1.getStrides(), el));
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp:271]: why do we need to pass byte strides explicitly? is it supposed that we test ROI?

File: inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp (hunk 5 above)
Comment (ilya-lavrenov) [reply inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp:299]: why do we need set_shape for outTensor ?

File: inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp [hunk 13]
```diff
@@ -176,6 +176,170 @@ TEST_P(OVInferRequestIOTensorTest, canInferWithGetOut) {
...
+
+    auto byteStrides = [](const ov::Strides& strides, const ov::element::Type& type) -> ov::Strides {
+        ov::Strides byte_strides(strides.size());
+        for (size_t i = 0; i < strides.size(); ++i)
+            byte_strides[i] = strides[i] * type.size();
+        return byte_strides;
+    };
+
+    ie::BlockingDesc blockingDesc1{{1, 4, 20, 20}, InferenceEngine::NCHW};
+    ov::runtime::Tensor tensor1(el, blockingDesc1.getBlockDims(), data, byteStrides(blockingDesc1.getStrides(), el));
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp:319]: don't understand why byteStrides are needed here. Seems no ROI is tested here

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp [hunk 14]
```diff
@@ -68,6 +69,94 @@ void OVInferRequestDynamicTests::TearDown() {
     function.reset();
 }
 
+TEST_P(OVInferRequestDynamicTests, InferDynamicNetwork) {
+    std::vector<ov::Shape> vectorShapes{{1, 4, 20, 20}, {1, 4, 20, 20}, {1, 4, 10, 10}};
+    bool checkOutputValues = !strcmp(function->get_friendly_name().c_str(), "Sin");
+    const std::string tensor_name = "input_tensor";
+    std::map<std::string, ov::PartialShape> shapes;
+    shapes[tensor_name] = { ov::Dimension(1, 5), ov::Dimension(1, 5), ov::Dimension(10, 40), ov::Dimension(10, 40) };
```
Comment (mandrono) [inline src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp:77]: looks like enough to pass as test parameter
`::testing::Values(getFunction3({{1,5}, {1, 5}, {10, 40}, {10, 40})),`

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp [hunk 15]
```diff
@@ -68,6 +69,94 @@ void OVInferRequestDynamicTests::TearDown() {
     function.reset();
 }
 
+TEST_P(OVInferRequestDynamicTests, InferDynamicNetwork) {
+    std::vector<ov::Shape> vectorShapes{{1, 4, 20, 20}, {1, 4, 20, 20}, {1, 4, 10, 10}};
+    bool checkOutputValues = !strcmp(function->get_friendly_name().c_str(), "Sin");
```
Comment (mandrono) [inline src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp:74]: maybe better to make it parameter of test?

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp [hunk 16]
```diff
@@ -68,6 +69,94 @@ void OVInferRequestDynamicTests::TearDown() {
...
+    // Create InferRequest
+    ov::runtime::InferRequest req;
+    for (auto& shape : vectorShapes) {
+        ov::runtime::Tensor inTensor = ov::test::utils::create_and_fill_tensor(element::f32, shape, 100, -50);
+        OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+        OV_ASSERT_NO_THROW(req.set_tensor("input_tensor", inTensor));
+        OV_ASSERT_NO_THROW(req.infer());
+        OV_ASSERT_NO_THROW(req.start_async());
+        OV_ASSERT_NO_THROW(req.wait());
+        if (checkOutputValues) {
```
Comment (mandrono) [inline src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp:90]: shouldn't we add accuracy comparison in other OVInferRequestDynamicTests?

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp [hunk 17]
```diff
@@ -68,6 +69,94 @@ void OVInferRequestDynamicTests::TearDown() {
...
+            ov::runtime::Tensor outTensor = req.get_tensor("Sin");
+            for (int i = 0; i < inTensor.get_size(); i++) {
+                ASSERT_LT(fabs(sin(inTensor.data<float>()[i]) - outTensor.data<float>()[i]), std::numeric_limits<float>::epsilon());
+            }
+        }
+    }
+}
+
+TEST_P(OVInferRequestDynamicTests, InferDynamicNetworkSetOutputTensorBeforeInfer) {
```
Comment (mandrono) [inline src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp:99]: could you also add test case when we set output tensor which was created on pre-allocated memory. 
ctor: `Tensor(const element::Type type, const Shape& shape, void* host_ptr, const Strides& strides = {});`
and check that data pointers are equal

File: inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp (hunk 5 above)
Comment (mandrono) [reply inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp]: I guess we need to test all possible scenario

**Comment by mandrono** [comment]
@egorshul could you also add test case for **static** model please?
1. complie_model with shape1
2. tensor = get_tensor()
3. tensor->set_shape(shape2) // shape2 != shape1
4. infer should raise exception

for input and output tensor

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp [hunk 18]
```diff
@@ -59,6 +60,16 @@ std::string OVInferRequestDynamicTests::getTestCaseName(testing::TestParamInfo<O
 void OVInferRequestDynamicTests::SetUp() {
     SKIP_IF_CURRENT_TEST_IS_DISABLED()
     std::tie(function, inOutShapes, targetDevice, configuration) = this->GetParam();
+    checkOutputValues = !strcmp(function->get_friendly_name().c_str(), "Sin");
```
Comment (mandrono) [inline src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp:63]: I guess better to add this flag into `OVInferRequestDynamicParams` instead comparison with function name

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp [hunk 19]
```diff
@@ -68,6 +79,164 @@ void OVInferRequestDynamicTests::TearDown() {
     function.reset();
 }
 
+TEST_P(OVInferRequestDynamicTests, InferStaticNetworkSetInputTensor) {
```
Comment (mandrono) [inline src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp:82]: maybe better to move static tests `to io_tensor.cpp` ?

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp [hunk 20]
```diff
@@ -68,6 +79,164 @@ void OVInferRequestDynamicTests::TearDown() {
...
+    const ov::Shape shape2 = inOutShapes[1].first;
+    std::map<std::string, ov::PartialShape> shapes;
+    shapes[function->inputs().back().get_any_name()] = shape1;
+    OV_ASSERT_NO_THROW(function->reshape(shapes));
+    // Load ov::Model to target plugins
+    auto execNet = ie->compile_model(function, targetDevice, configuration);
+    // Create InferRequest
+    ov::runtime::InferRequest req;
+    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+    // Get input_tensor
```
Comment (mandrono) [inline src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp:93]:
```suggestion
    ov::runtime::InferRequest req;
    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
    OV_ASSERT_NO_THROW(req.infer());
    // Get input_tensor
```

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp [hunk 21]
```diff
@@ -68,6 +79,164 @@ void OVInferRequestDynamicTests::TearDown() {
...
+    const ov::Shape shape2 = inOutShapes[1].first;
+    std::map<std::string, ov::PartialShape> shapes;
+    shapes[function->inputs().back().get_any_name()] = shape1;
+    OV_ASSERT_NO_THROW(function->reshape(shapes));
+    // Load ov::Model to target plugins
+    auto execNet = ie->compile_model(function, targetDevice, configuration);
+    // Create InferRequest
+    ov::runtime::InferRequest req;
+    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+    // Get output_tensor
```
Comment (mandrono) [inline src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp:112]:
```suggestion
    ov::runtime::InferRequest req;
    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
    OV_ASSERT_NO_THROW(req.infer());
    // Get output_tensor
```

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp [hunk 22]
```diff
@@ -68,6 +79,164 @@ void OVInferRequestDynamicTests::TearDown() {
...
+        OV_ASSERT_NO_THROW(req.infer());
+        OV_ASSERT_NO_THROW(req.start_async());
+        OV_ASSERT_NO_THROW(req.wait());
+        if (checkOutputValues) {
+            ASSERT_EQ(true, checkOutput(req.get_tensor(function->inputs().back().get_any_name()), req.get_tensor("Sin")));
+        }
+    }
+}
+
+TEST_P(OVInferRequestDynamicTests, InferDynamicNetworkSetOutputTensorBeforeInfer) {
```
Comment (mandrono) [inline src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp:154]:
```suggestion
TEST_P(OVInferRequestDynamicTests, InferDynamicNetworkSetUnexpectedOutputTensorBeforeInfer) {
```

File: src/tests/functional/plugin/shared/include/behavior/ov_infer_request/infer_request_dynamic.hpp [hunk 23]
```diff
@@ -50,11 +50,13 @@ class OVInferRequestDynamicTests : public testing::WithParamInterface<OVInferReq
     void SetUp() override;
 
     void TearDown() override;
+    bool checkOutput(const ov::runtime::Tensor& in, const ov::runtime::Tensor& out);
```
Comment (mandrono) [inline src/tests/functional/plugin/shared/include/behavior/ov_infer_request/infer_request_dynamic.hpp:53]:
```suggestion
    bool checkOutput(const ov::runtime::Tensor& in, const ov::runtime::Tensor& actual);
```

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp [hunk 24]
```diff
@@ -68,6 +79,126 @@ void OVInferRequestDynamicTests::TearDown() {
...
+    ov::runtime::InferRequest req;
+    for (auto& shape : vectorShapes) {
+        ov::runtime::Tensor inTensor = ov::test::utils::create_and_fill_tensor(element::f32, shape, 100, -50);
+        OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+        OV_ASSERT_NO_THROW(req.set_tensor("input_tensor", inTensor));
+        OV_ASSERT_NO_THROW(req.infer());
+        OV_ASSERT_NO_THROW(req.start_async());
+        OV_ASSERT_NO_THROW(req.wait());
+        if (checkOutputValues) {
+            ASSERT_EQ(true, checkOutput(req.get_tensor(function->inputs().back().get_any_name()), req.get_tensor("Sin")));
```
Comment (mandrono) [inline src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp:111]:
```suggestion
            ASSERT_TRUE(true, checkOutput(req.get_tensor(function->inputs().back().get_any_name()), req.get_tensor("Sin")));
```

File: src/tests/functional/plugin/cpu/shared_tests_instances/behavior/ov_infer_request/infer_request_dynamic.cpp [hunk 25]
```diff
@@ -64,6 +64,18 @@ std::shared_ptr<ngraph::Function> getFunction2() {
     return std::make_shared<ngraph::Function>(concat, params, "SplitAddConcat");
 }
 
+std::shared_ptr<ngraph::Function> getFunction3(ov::PartialShape shape) {
+    const ngraph::element::Type_t ngPrc = ngraph::element::Type_t::f32;
```
Comment (mandrono) [inline src/tests/functional/plugin/cpu/shared_tests_instances/behavior/ov_infer_request/infer_request_dynamic.cpp:68]: let's align with others - `getFunction3` also should create static function

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp (hunk 18 above)
Comment (mandrono) [reply src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp:63]: As far as I remember we need to update instance for other plug-ins to pass `checkOutputValues` as test param. So let's do it.

Comment (mandrono) [reply src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp:63]: Also can we pass `std::function` that will execute reference? It looks more flexible. For `checkOutputValues == false` `std::function` should throw exception

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp [hunk 26]
```diff
@@ -61,13 +62,144 @@ void OVInferRequestDynamicTests::SetUp() {
...
+    // Create InferRequest
+    ov::InferRequest req;
+    const std::string outputname = function->outputs().back().get_any_name();
+    for (auto& shape : vectorShapes) {
+        ov::runtime::Tensor inTensor = ov::test::utils::create_and_fill_tensor(element::f32, shape, 100, -50);
+        OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+        OV_ASSERT_NO_THROW(req.set_tensor("input_tensor", inTensor));
+        OV_ASSERT_NO_THROW(req.infer());
+        OV_ASSERT_NO_THROW(req.start_async());
+        OV_ASSERT_NO_THROW(req.wait());
```
Comment (mandrono) [inline src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp:118]: should be `infer` or `start_async` + `wait`. not both of them

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp [hunk 27]
```diff
@@ -61,13 +62,144 @@ void OVInferRequestDynamicTests::SetUp() {
...
+    const std::string outputname = function->outputs().back().get_any_name();
+    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+    tensor = ov::test::utils::create_and_fill_tensor(element::f32, refShape, 100, -50);
+    OV_ASSERT_NO_THROW(req.set_tensor("input_tensor", tensor));
+    float ptr[5000];
+    ov::runtime::Tensor otensor(element::f32, refOutShape, ptr);
+    OV_ASSERT_NO_THROW(req.set_tensor(outputname, otensor));
+    OV_ASSERT_NO_THROW(req.infer());
+    OV_ASSERT_NO_THROW(req.start_async());
+    OV_ASSERT_NO_THROW(req.wait());
```
Comment (mandrono) [inline src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_request_dynamic.cpp:171]: should be `infer` or `start_async` + `wait`. not both of them

**Review by mandrono — APPROVED** [review]
Please fix remaining comments. In other LGTM.

File: src/tests/functional/plugin/shared/include/behavior/ov_infer_request/infer_request_dynamic.hpp [hunk 28]
```diff
@@ -35,10 +35,10 @@ namespace test {
...
 using OVInferRequestDynamicParams = std::tuple<
-        std::shared_ptr<Model>,                                         // ov Model
+        std::shared_ptr<Model>,                                            // ov Model
         std::vector<std::pair<std::vector<size_t>, std::vector<size_t>>>,  // input/expected output shapes per inference
         std::string,                                                       // Device name
-        ov::AnyMap                                                  // Config
+        ov::AnyMap                                                         // Config
 >;
 
 class OVInferRequestDynamicTests : public testing::WithParamInterface<OVInferRequestDynamicParams>,
```
[the remark this answers is not in the document]
Comment (egorshul) [reply src/tests/functional/plugin/shared/include/behavior/ov_infer_request/infer_request_dynamic.hpp:45]: Fixed in 99041afb13b4691ea4c9f9c33c1d06bb8a91edc1

**Comment by zhangYiIntel** [comment]
Close this PR and migrate this change to PR #11840

**Pull request closed without merging**